pkg/controllers/jobset_controller.go

```diff
@@ -675,8 +675,12 @@ func labelAndAnnotateObject(obj metav1.Object, js *jobset.JobSet, rjob *jobset.R
 
 	// Only set exclusive key label/annotation on jobs/pods if the parent JobSet
 	// is using exclusive placement.
-	if _, exists := js.Annotations[jobset.ExclusiveKey]; exists {
-		annotations[jobset.ExclusiveKey] = js.Annotations[jobset.ExclusiveKey]
+	if topologyDomain, exists := rjob.Template.Annotations[jobset.ExclusiveKey]; exists {
```

Comment:
same thing here, check both `js.Annotations[jobset.ExclusiveKey]` and `rjob.Template.Annotations[jobset.ExclusiveKey]` and